### ISSUE TITLE
Many problems with TypeScript 2.5.2 and Visual Studio 2015 (Update 3)

### ISSUE THREAD

**Issue by KarloX2** [Microsoft/TypeScript#18571]
**TypeScript Version:**  2.5.2

I recently updated my VS2015 to TypeScript 2.5.2 and started working with package.json to load some type definition (d.ts) modules. I'm facing a lot of problems, very hard to overcome. Very unstable.

- Though I managed to get rid of all errors (as displayed in the Error List window), .js output files are not being generated when I save my .ts files, unless I uncheck "Do not emit outputs if any errors are reported" on the "TypeScript Build" tab card in my projects settings.
- Even with that flag being unchecked, output files are sometimes generated (just fine) but sometimes not. When it gets stuck, the only help is to close Visual Studio, clean bin/obj folders, restart, rebuild, pray, and retry.
- Sometimes, out of a sudden, error messages like "Cannot find type definition files for 'jquery'" appear in the error list. However, the definition file is present and I can work with it. To get rid of these Messages I need to restart VS, a re-built won't raise these errors, but generating a .ts file will (but sometimes won't).
- I have multiple projects in my solution. Sometimes generating outputs stop at one project, while it still works in another. I can't find a pattern.

Sorry, I'm unable to come up with clear steps to reproduce.

- Does anybody experience the same?
- Are there already related bugs reported?
- Is TypeScript Version 2.5.2 with VS2015 known to be unstable?

Thanks!

**Comment by mhegazy** [comment]
is this happening for a single project or all projects? do you see these on other installation/machines?

**Comment by KarloX2** [comment]
I'm sorry I cannot answer these questions. I don't habe another VS 2015 installation of this kind at hand and I don't have other projects using TypeScript in this way yet.

However, I installed VS2017.3 and with this the behavior seems to be much better (as far as I can tell until now).

**Comment by mhegazy** [comment]
> I'm sorry I cannot answer these questions. I don't habe another VS 2015 installation of this kind at hand and I don't have other projects using TypeScript in this way yet.

do you see it if you create an empty project?

**Comment by KarloX2** [comment]
I started a new MVC Web project with VS2015.3 + TypeScript 2.5.2.

First, I saw similar strange issues, but I was now able to work around by adding a tsconfig.json file to that project. (It took very long for me to understand that this is possible and that the tsconfig.json file will then outvote all typescript settings made in the .csproj file previously).

I think the problem had to do with the `@types/jquery` (V3.2.12) type definitions referencing `Iterable<TElement>` not defined for ES5 (which is my target).

So I can now close this request. Thanks for your support!

For everyone facing similar problems, here's my tsconfig.json:

```
{
  "compileOnSave": true,
  "compilerOptions": {
    "declaration": false,
    "lib": [ "dom", "es2015" ],
    "module": "commonjs",
    "noImplicitAny": false,
    "noEmitOnError": true,
    "removeComments": true,
    "sourceMap": true,
    "suppressImplicitAnyIndexErrors": false,
    "target": "es5"
  },
  "exclude": [
    "node_modules"
  ]
}
